pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/EndpointImpl.kt

```diff
@@ -7,11 +7,13 @@ import com.pubnub.api.v2.PNConfigurationOverride
 import com.pubnub.internal.v2.PNConfigurationImpl
 
 abstract class EndpointImpl<OUTPUT>(private val endpoint: HasOverridableConfig) : Endpoint<OUTPUT> {
-    override fun overrideConfiguration(action: PNConfigurationOverride.Builder.() -> Unit) {
+    override fun overrideConfiguration(action: PNConfigurationOverride.Builder.() -> Unit): Endpoint<OUTPUT> {
         endpoint.overrideConfiguration(PNConfigurationImpl.Builder(endpoint.configuration).apply(action).build())
+        return this
     }
 
-    override fun overrideConfiguration(configuration: PNConfiguration) {
+    override fun overrideConfiguration(configuration: PNConfiguration): Endpoint<OUTPUT> {
```

Comment:
I would add/modify test to demonstrate usage of kind of builder pattern way of using overrideConfiguration

Reply:
done